viewer/packages/360-images/src/icons/IconCollection.ts

```diff
@@ -0,0 +1,211 @@
+  }
+
+  private initializeImage360Icons(
+    points: Vector3[],
+    sceneHandler: SceneHandler,
+    onBeforeSceneRendered: EventTrigger<BeforeSceneRenderedDelegate>
+  ): Image360Icon[] {
+    return points.map(
+      point =>
+        new Image360Icon(
+          point,
+          this._hoverIconTexture,
+          sceneHandler,
+          this.MIN_PIXEL_SIZE,
+          this.MAX_PIXEL_SIZE,
+          onBeforeSceneRendered
+        )
+    );
+  }
+
+  public dispose(): void {
+    this._sceneHandler.removeCustomObject(this._clusterSprites);
+    this._sceneHandler.removeCustomObject(this._iconsSprite);
+    this._clusterSprites.dispose();
+    this._iconsSprite.dispose();
+    this._onRenderTrigger.unsubscribeAll();
+    this._sharedTexture.dispose();
+  }
+
+  private createClusterTexture(): CanvasTexture {
+    const canvas = document.createElement('canvas');
+    const textureSize = this.MAX_PIXEL_SIZE;
+    canvas.width = textureSize;
+    canvas.height = textureSize;
+
+    const halfTextureSize = textureSize / 2;
+
+    const context = canvas.getContext('2d')!;
+    drawClusterRings();
+    drawInnerCircle();
+    drawOuterCircle();
+
+    return new CanvasTexture(canvas);
+
+    function drawClusterRings() {
+      context.beginPath();
+      context.lineWidth = textureSize / 22;
+      context.strokeStyle = '#FFFFFF';
+      context.arc(halfTextureSize, halfTextureSize, (halfTextureSize * 36) / 44, 0, 2 * Math.PI);
+      context.stroke();
+
+      context.beginPath();
+      context.lineWidth = textureSize / 22;
+      context.strokeStyle = '#FFFFFF';
+      context.arc(halfTextureSize, halfTextureSize, halfTextureSize - context.lineWidth / 2, 0, 2 * Math.PI);
+      context.stroke();
+    }
+
+    function drawOuterCircle() {
+      context.beginPath();
+      context.lineWidth = textureSize / 22;
+      context.strokeStyle = '#FFFFFF';
+      context.arc(halfTextureSize, halfTextureSize, (halfTextureSize * 30) / 44, 0, 2 * Math.PI);
+      context.stroke();
+    }
+
+    function drawInnerCircle() {
+      context.beginPath();
+      context.lineWidth = textureSize / 11;
+      context.strokeStyle = 'rgba(255, 255, 255, 0.75)';
+      context.arc(halfTextureSize, halfTextureSize, (halfTextureSize * 24) / 44, 0, 2 * Math.PI);
+      context.shadowColor = 'red';
+      context.stroke();
+    }
+  }
+
+  private createOuterRingsTexture(): CanvasTexture {
+    const canvas = document.createElement('canvas');
+    const textureSize = this.MAX_PIXEL_SIZE;
+    canvas.width = textureSize;
+    canvas.height = textureSize;
+
+    const context = canvas.getContext('2d')!;
+    drawInnerCircle();
+    drawOuterCircle();
+
+    return new CanvasTexture(canvas);
+
+    function drawOuterCircle() {
+      context.beginPath();
+      context.lineWidth = textureSize / 16;
+      context.strokeStyle = '#FFFFFF';
+      context.arc(textureSize / 2, textureSize / 2, textureSize / 2 - context.lineWidth / 2 - 2, 0, 2 * Math.PI);
```

Comment:
It would be good to we have `const arcCenter = textureSize * 0.5` and use it in `context.arc()`